File: backend/src/main/java/com/dnd/backend/dto/DndCharacterDTO.java
package com.dnd.backend.dto;

import com.dnd.backend.constant.CharacterClass;
import com.dnd.backend.constant.CharacterRace;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.Builder;

@Builder
public record DndCharacterDTO(
    Long id,
    @NotNull(message = "First name is required")
    String firstName,
    @NotNull(message = "Last name is required")
    String lastName,
    CharacterClass dndClass,
    CharacterRace race,
    @Min(15)
    int age,
    boolean canPerformMagic,
    MainSkillDTO skill
) {}

File: frontend/src/app/shared/dtos/dnd-character.dto.ts
import { DndClass } from "../enums/dnd-class.enum";
import { Race } from "../enums/race.enum";
import { SkillDto } from "./skill.dto";

export interface DndCharacterDto {
    readonly id: number | null,
    readonly firstName: string,
    readonly lastName: string,
    readonly dndClass: DndClass,
    readonly race: Race,
    readonly age: number,
    readonly canPerformMagic: boolean,
    readonly skill: SkillDto | null,
};